docs/python/tutorial-azure-functions.md

```diff
@@ -0,0 +1,348 @@
+> ```
+
+### Verify prerequisites
+
+To verify that all the Azure Functions tools are installed, select the **Terminal: Create New Integrated Terminal** command from the Command Palette, then run the command `func`. You should see output indicating that the core tools are present:
+
+![Checking Azure Functions core tools prerequisites](images/functions/check-prereqs.png)
+
+## Create the Function
+
+1. Code for Azure Functions is managed within a Function "project," which you create first before creating the function itself. In **Azure: Function** explorer, select the **New Project** command icon, or open the Command Palette and select **Azure Functions: Create New Project**.
+
+    ![Create new project button in the Functions explorer](images/functions/project-create-new.png)
+
+1. In the prompts that follow:
+
+    - Specify the folder for the project (the default is the current folder open in VS Code)
+    - Select **Python** for the language.
+    - Select **HTTP trigger** for the template. A function that uses an HTTP trigger is run whenever there's an HTTP request made to the function's endpoint. (You can see that there are a variety of other triggers for Azure Functions. To learn more, see [What can I do with Functions?](https://docs.microsoft.com/azure/azure-functions/functions-overview#what-can-i-do-with-functions) in the Azure documentation.)
+    - Name your function "HttpExample" (rather than accepting the default "HTTPTrigger") to distinguish the function itself from the trigger. This name is used for a subfolder that contains the function's code along with configuration data, and also defines the name of the HTTP endpoint.
+    - Select **Anonymous** for the authorization level, which makes the function publicly accessible to anyone.
+
+1. After a short time you see a message that the new project was created. In the **Explorer** you see the subfolder created for the function, and VS Code opens the `__init__.py` file that contains the default function code:
+
+    ![Result of creating a new Python functions project](images/functions/project-create-results.png)
+
+1. When `__init__.py` opens, VS Code might tell you that you don't have a Python interpreter selected. In that case, use the **Python: Select Interpreter** command from the Command Palette and select the virtual environment in the local `.env` folder (which was created as part of the project).
+
+    ![Selecting the virtual environment created with the project](images/functions/select-venv-interpreter.png)
+
+> **Tip**: Whenever you want to create another function in the same project, use the **Create Function** command in the **Azure: Functions** explorer, or use the **Azure Functions: Create Function** command from the Command Palette. Both commands prompt you for a function name (which is the name of the endpoint), then creates a subfolder with the default files.
+>
+> ![New Function command in the Azure: Functions explorer](images/functions/function-create-new.png)
+
```

Comment:
Also requires that you manually install the `azure.functions` module (!) which is not `pip install`ed by the project creation phase. This can just be done directly in the terminal once the virtualenv is loaded:

```
pip install azure.functions
```

Reply:
Looking into this some more, as part of running the debug config later in the article: Even then it reports an error about not being able to import azure.functions:

![image](https://user-images.githubusercontent.com/169061/56396930-24471c80-61f6-11e9-9a3c-5bf3e425182e.png)

Clicked 'continue debugging anyway' - and it worked (????).

Reply:
I think this might be a bug somewhere. I never did an explicit pip install and things worked fine...

Reply:
Also, https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-python doesn't say anything about that, so I think the issue is in the language server and not in Functions.

Reply:
Also, Rabeea didn't report this as an issue, so I'm going to leave it be for now.